api/src/main/java/org/apache/iceberg/types/Conversions.java

```diff
@@ -71,6 +71,12 @@ public static Object fromPartitionString(Type type, String asString) {
         return new BigDecimal(asString);
       case DATE:
         return Literal.of(asString).to(Types.DateType.get()).value();
+      case TIMESTAMP:
+        if (!asString.contains("T")) {
+          return java.sql.Timestamp.valueOf(asString).getTime() * 1000;
```

Comment:
The path is something like this `ts=2021-01-01 00:00:00.999`

Reply:
Will time zones be confused?

Reply:
My understanding is that the timezone has already been taken care of. For example, if I have row
```
1, "John Doe", "hr", toTimestamp("2021-01-01T00:00:00.999999999Z"
```
The partition path is actually `ts=2020-12-31 16:00:00.999`.

Reply:
I was wondering here if we can find do the same inversion that's in the hive code for parsing this?

Reply:
I notice that the test only creates one type of timestamp string so do we only need to cover that?

Reply:
We need to be _very_ strict with what we parse and accept here and delegating this work to `java.sql.Timestamp` seems like a good way to introduce formats that we don't intend to support. I would rather use what is built into `StringLiteral` and supply different formats. Can you look at using `OffsetDateTime.parse` and a format for this instead?